File: 2415_reverse_odd_levels_of_binary_tree.cpp
class Solution {
public:
    TreeNode* reverseOddLevels(TreeNode* a) {
        queue<TreeNode*> b;
        b.push(a);
        bool c = false;
        while (!b.empty()) {
            int d = b.size();
            vector<TreeNode*> e;
            for (int f = 0; f < d; f++) {
                TreeNode* g = b.front();
                b.pop();
                if (!g) continue;
                e.push_back(g);
                b.push(g->left);
                b.push(g->right);
            }
            if (c) {
                int h = 0, i = e.size() - 1;
                while (h < i) swap(e[h++]->val, e[i--]->val);
            }
            c = !c;
        }
        return a;
    }
};

File: 2415_reverse_odd_levels_of_binary_tree.java
class Solution {
    public TreeNode reverseOddLevels(TreeNode z) {
        Queue<TreeNode> y = new LinkedList<>();
        y.add(z);
        boolean x = false;
        while (!y.isEmpty()) {
            List<TreeNode> w = new ArrayList<>();
            for (int v = y.size(); v-- > 0;) {
                TreeNode u = y.poll();
                if (u == null) continue;
                w.add(u);
                y.add(u.left);
                y.add(u.right);
            }
            if (x) {
                for (int t = 0, s = w.size() - 1; t < s; t++, s--) {
                    int tmp = w.get(t).val;
                    w.get(t).val = w.get(s).val;
                    w.get(s).val = tmp;
                }
            }
            x = !x;
        }
        return z;
    }
}
